smdebug/core/state_store.py

```diff
@@ -74,14 +74,24 @@ def _read_states_file(self):
 
     def is_checkpoint_updated(self):
         """
-        Check whether the folder in which checkpoints are stored got updated.
-        Update to that folder indicates, user attempted to store the new checkpoint in that directory.
+        Check whether new checkpoint files got added or existing checkpoint files that are
+        stored got updated.
         """
         if self._checkpoint_dir is not None:
-            checkpoint_timestamp = max(
-                os.path.getmtime(child) for child, _, _ in os.walk(self._checkpoint_dir)
-            )
-            if checkpoint_timestamp > self._checkpoint_update_timestamp:
+            timestamps = [
+                os.path.getmtime(child)
+                for child, _, _ in os.walk(self._checkpoint_dir)
+                if child != self._checkpoint_dir
+            ]
+            _, _, files = next(os.walk(self._checkpoint_dir))
+            timestamps += [
+                os.path.getmtime(os.path.join(self._checkpoint_dir, file))
+                for file in files
+                if file != METADATA_FILENAME
+            ]
+            if not timestamps:
```

Comment:
I think we need a simple way to recursively get all the files.  
import os
>>> for child, r, c in os.walk("/tmp/newlogsRunTest/"):
...   for f in c:
...     if f != "haha":
...       print(child,"/",f)

Also see here- 
https://stackoverflow.com/questions/2186525/how-to-use-glob-to-find-files-recursively